component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -0,0 +1,201 @@
+ */
+
+@Extension(
+        name = "websocket",
+        namespace = "sink",
+        description = "A Siddhi application can be configured to publish events via the Websocket transport by " +
+                "adding the @Sink(type = ‘Websocket’) annotation at the top of an event stream definition.",
+        parameters = {
+                @Parameter(
+                        name = "url",
+                        description = "The URL of the remote endpoint.\n" +
+                                "The url scheme should be either ‘ws’ or ‘wss’.",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "sub.protocol",
+                        description = "The negotiable sub-protocol if server is asking for it.\n" +
+                                "The sub.protocol should adhere to `subprotocol1, subprotocol2,...` format.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"
+                ),
+                @Parameter(
+                        name = "headers",
+                        description = "Any specific headers which need to send to the server.\n" +
+                                "The headers should adhere to `'key1:value1', 'key2:value2',...` format.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"
+                ),
+                @Parameter(
+                        name = "idle.timeout",
+                        description = "Idle timeout of the connection",
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = "-1"
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "@Sink(type = ‘websocket’, url = 'ws://localhost:8025/abc', \n" +
+                                "   @map(type='xml'))\n" +
+                                "define stream Foo (attribute1 string, attribute2 int);",
+                        description = "" +
+                                "A sink of type 'websocket' has been defined.\n" +
+                                "All events arriving at Foo stream via websocket will be sent " +
+                                "to the url ws://localhost:8025/abc."
+                )
+        }
+)
+
+public class WebSocketSink extends Sink {
+    private StreamDefinition streamDefinition;
+    private String url;
+    private String subProtocol;
+    private String headers;
+    private String idleTimeoutString;
+    private int idleTimeout;
+    private WebSocketClientConnectorListener connectorListener;
+    private HandshakeFuture handshakeFuture = null;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, ByteBuffer.class};
```

Comment:
instead of returning a new array, lets return a constant
```
private static final Class[] SUPPORTED_INPUT_EVENT_CLASSES = new Class[]{String.class, ByteBuffer.class};
```
in the method, return `SUPPORTED_INPUT_EVENT_CLASSES` constant.

Applies for the `getSupportedDynamicOptions` method as well.

Reply:
fixed in b5c9ede